src/main/java/bio/terra/service/resourcemanagement/google/GoogleProjectService.java

```diff
@@ -461,4 +461,34 @@ private ServiceUsage serviceUsage() throws IOException, GeneralSecurityException
         }
         return operation;
     }
+
+    public void updateProjectsBillingAccount(BillingProfileModel billingProfileModel) {
+        List<GoogleProjectResource> projects = resourceDao
+            .retrieveProjectsByBillingProfileId(UUID.fromString(billingProfileModel.getId()));
+
+        if (projects.size() == 0) {
+            logger.info("No projects attached to billing profile so nothing to update.");
+            return;
+        }
+
+        projects.stream().forEach(project -> {
+            logger.info("Updating billing profile id {} in google project {}",
+                billingProfileModel.getBillingAccountId(), project.getGoogleProjectId());
+            updateBillingProfile(
+                project.getGoogleProjectNumber(), project.getGoogleProjectId(), billingProfileModel);
+        });
+    }
+
+    public void updateBillingProfile(String googleProjectNumber,
+                                     String googleProjectId,
+                                     BillingProfileModel billingProfile) {
+        GoogleProjectResource googleProjectResource =
+            new GoogleProjectResource()
+                .profileId(UUID.fromString(billingProfile.getId()))
+                .googleProjectId(googleProjectId)
+                .googleProjectNumber(googleProjectNumber);
+
+        // The billing profile has already been authorized so we do no further checking here
+        billingService.assignProjectBilling(billingProfile, googleProjectResource);
```

Comment:
Do we know anything about possible errors here or the need for retries? (We would probably do it in billingService, but thought of it here...)